tests/raftstore_cases/test_merge.rs

```diff
@@ -357,13 +357,25 @@ fn test_node_merge_brain_split() {
     cluster.must_transfer_leader(1, new_peer(1, 1));
     let pd_client = Arc::clone(&cluster.pd_client);
     let region = pd_client.get_region(b"k1").unwrap();
+
     cluster.must_split(&region, b"k2");
+    cluster.must_put(b"k11", b"v11");
+    cluster.must_put(b"k21", b"v21");
+    for i in 1..4 {
+        util::must_get_equal(&cluster.get_engine(i), b"k11", b"v11");
+        util::must_get_equal(&cluster.get_engine(i), b"k21", b"v21");
+    }
+
     let left = pd_client.get_region(b"k1").unwrap();
+    let right = pd_client.get_region(b"k3").unwrap();
+
+    // Splited regions' leaders could be at store 3, so transfer them to peer 1.
```

Comment:
`The split regions'`

Reply:
I believe merge should work in this case.

Reply:
If the leader of one region is at store 3 when store 3 is isolated, the progress of `peer (3, 3)` of new leader will be `0`. So the merge can't be performed until the isolation is removed.